structuretoolkit/build/mesh.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+import warnings
+from ase.atoms import Atoms
+
+
+def create_mesh(
+    structure: Atoms,
+    n_mesh: int|list = 10,
+    density: float|None = None,
+    endpoint: bool = False
+):
+    """
+    Create a mesh based on a structure
+
+    Args:
+        structure (ase.atoms.Atoms): ASE Atoms
+        n_mesh (int): Number of grid points in each direction. If one number
+            is given, it will be repeated in every direction (i.e. n_mesh = 3
+            is the same as n_mesh = [3, 3, 3])
+        density (float): Density of grid points. Ignored when n_mesh is not
+            None
+        endpoint (bool): Whether both the edges get separate points or not.
+            cf. endpoint in numpy.linspace
+
+    Returns:
+        (n, n, n, 3)-array: mesh
+    """
+    if n_mesh is None:
+        if density is None:
+            raise ValueError("either n_mesh or density must be specified")
+        n_mesh = np.rint(np.linalg.norm(structure.cell, axis=-1) / density).astype(int)
+    elif density is not None:
+        warnings.warn("As n_mesh is not `None`, `density` is ignored")
```

Comment:
Why not fail hard? If they've bothered to specify a density, they've either forgotten to turn off the `n_mesh` and probably don't want this to run and would appreciate it not to, or they've intentionally specified both because they misunderstand the basics and we should slap them in the face with an error anyhow.

Reply:
I changed it